Katello/katello
# fixes #25224 - show arch from product as backup

If there's no basearch substitution, we can check and see if Candlepin knows the arch for the product. Under the new CDN structure, repository sets will only have 1 arch and no substitution variable.  So, this PR makes a differentiation between the display architecture in the UI, and an architecture substitution.  I'm not sure if this is the best approach, maybe someone familiar with the repos page has a better idea of how to fix this.


To test -

1. Use a manifest that has beta products in it, and hit the stage CDN
2. Notice repo sets that have arch in the name show "Undefined" when trying to enable them
3. Apply this patch
4. Notice that it correctly shows the architecture in the  list
